docs/reference/indices/diskusage.asciidoc

```diff
@@ -0,0 +1,184 @@
+[[indices-disk-usage]]
+=== Analyze index disk usage API
+++++
+<titleabbrev>Analyze index disk usage</titleabbrev>
+++++
+
+experimental[]
+
+Analyzes the disk usage of each field of an index or data stream.
+This API might not support indices created in previous {es} versions.
+The result of a small index can be inaccurate as some parts of an index
+might not be analyzed by the API.
+
+[source,console]
+--------------------------------------------------
+POST /my-index-000001/_disk_usage?run_expensive_tasks=true
+--------------------------------------------------
+// TEST[setup:messages]
+
+[[analyze-index-disk-usage-api-request]]
+==== {api-request-title}
+
+`POST /<target>/_disk_usage`
+
+[[analyze-index-disk-usage-api-request-prereqs]]
+==== {api-prereq-title}
+
+* If the {es} {security-features} are enabled, you must have the `manage`
+<<privileges-list-indices,index privilege>> for the target index, data stream,
+or alias.
+
+[[analyze-index-disk-usage-api-path-params]]
+==== {api-path-parms-title}
+
+`<target>`::
+(Required, string) Comma-separated list of data streams, indices, and aliases
+used to limit the request. It's recommended to execute this API with a single
+index (or the latest backing index of a data stream) as the API consumes
+resources significantly.
+
+[[analyze-index-disk-usage-api-query-params]]
+==== {api-query-parms-title}
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
++
+Defaults to `true`.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=expand-wildcards]
++
+Defaults to `open`.
+
+`flush`::
+(Optional, Boolean) If `true`, the API performs a flush before analysis. If
+`false`, the response may not include uncommitted data. Defaults to `true`.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailable]
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=timeoutparms]
+
+`run_expensive_tasks`::
+(Required, Boolean) Analyzing field disk usage is resource-intensive. To use the
+API, this parameter must be set to `true`. Defaults to `false`.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=wait_for_active_shards]
+
+
+[[analyze-index-disk-usage-api-example]]
+==== {api-examples-title}
+
+[source,console]
+--------------------------------------------------
+POST /my-index-000001/_disk_usage?run_expensive_tasks=true
+--------------------------------------------------
+// TEST[setup:messages]
+
+The API returns:
+
+[source,console-response]
+--------------------------------------------------
+{
+    "_shards": {
+        "total": 1,
+        "successful": 1,
+        "failed": 0
+    },
+    "my-index-000001": {
+        "store_size": "929mb", <1>
+        "store_size_in_bytes": 974192723,
+        "all_fields": {
+            "total": "928.9mb", <2>
+            "total_in_bytes": 973977084,
+            "inverted_index": {
```

Comment:
should we return an object for all data structures for consistency?

Reply:
At least in the field usage API I'm inclined not to, as many of these would just have a single value below them. Here there would be two (X and X_in_bytes), so it might just make this overly hierarchical.

I have to admit that I'm not 100% happy with the `inverted_index` section either, but expect our thinking to evolve as we add more details to this API here in follow-ups.